akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -11,20 +11,20 @@ import akka.http.impl.util.JavaMapping.Implicits._
 
 final case class HttpChallenge(scheme: String, realm: String,
                                params: Map[String, String] = Map.empty) extends jm.headers.HttpChallenge with ValueRenderable {
-
   def render[R <: Rendering](r: R): r.type = {
-    r ~~ scheme ~~ " realm=" ~~#! realm
+    r ~~ scheme
+    if (realm != null) r ~~ " realm=" ~~#! realm
     if (params.nonEmpty) params.foreach { case (k, v) ⇒ r ~~ ',' ~~ k ~~ '=' ~~# v }
     r
   }
-
   /** Java API */
   def getParams: util.Map[String, String] = params.asJava
 }
 
-object HttpChallenges {
```

Comment:
no, we can not break this - binary compatibility is paramount.
Please bring back the entire object